pkg/api/roles.go

```diff
@@ -50,157 +50,189 @@ var (
-				Description: "Read current organization and its quotas.",
-				Permissions: []accesscontrol.Permission{
-					{
-						Action: ActionOrgsRead,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsQuotasRead,
-						Scope:  ScopeOrgCurrent,
-					},
-				},
+		Grants: []string{string(models.ROLE_VIEWER)},
+	}
+
+	datasourcesCompatibilityReaderRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:datasources:compatibility:querier",
+			DisplayName: "Data source compatibility querier",
+			Description: "Query data sources when data source permissions are not in use",
+			Permissions: []accesscontrol.Permission{
+				{Action: ActionDatasourcesQuery},
 			},
-			Grants: []string{string(models.ROLE_VIEWER)},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     1,
-				Name:        "fixed:current:org:writer",
-				Description: "Read current organization, its quotas, and its preferences. Write current organization and its preferences.",
-				Permissions: []accesscontrol.Permission{
-					{
-						Action: ActionOrgsRead,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsQuotasRead,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsPreferencesRead,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsWrite,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsPreferencesWrite,
-						Scope:  ScopeOrgCurrent,
-					},
+		Grants: []string{string(models.ROLE_VIEWER)},
+	}
+
+	currentOrgReaderRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:current.org:reader",
+			DisplayName: "Current Organization reader",
+			Description: "Read the current organization, such as its ID, name, address, or quotas.",
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: ActionOrgsRead,
+					Scope:  ScopeOrgCurrent,
+				},
+				{
+					Action: ActionOrgsQuotasRead,
+					Scope:  ScopeOrgCurrent,
 				},
 			},
-			Grants: []string{string(models.ROLE_ADMIN)},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     1,
-				Name:        "fixed:orgs:writer",
-				Description: "Create, read, write, or delete an organization. Read or write an organization's quotas.",
-				Permissions: []accesscontrol.Permission{
-					{Action: ActionOrgsCreate},
-					{
-						Action: ActionOrgsRead,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsWrite,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsDelete,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsQuotasRead,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsQuotasWrite,
-						Scope:  ScopeOrgsAll,
-					},
+		Grants: []string{string(models.ROLE_VIEWER)},
+	}
+
+	currentOrgWriterRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
```

Comment:
Depending on who merges first, we'll have to bump the org roles versions number again ;)